### ISSUE TITLE
TSQL PIVOT construction PRS error - small bug withowt AS keyword

### ISSUE THREAD

**Issue by tkachenkomaria244** [sqlfluff/sqlfluff#1730]
SQL code below doesn't cause parsing errors (TSQL dialect):
```
	from
		table1 as t1
	pivot 
		(max(value) for rn in([1], [2], [3]) ) as pvt
```
but the same withowt "as" - does:
```
	from
		table1 as t1
	pivot 
		(max(value) for rn in([1], [2], [3]) ) pvt
```

## Expected Behaviour
no PRS errors
## Observed Behaviour
```
==== parsing violations ====
L:  27 | P:   2 |  PRS | Line 27, Position 2: Found unparsable section: 'pivot
                       | \n\t\t(max(value) for rn in([1], [2],...'
```

## Dialect
tsql

## Version
SQLFluff 0.6.7 - 0.7.1 (I suppose the same was in previous versions but didn't check)
Python 3.9.6

## Configuration
```
[sqlfluff]
exclude_rules = L007,L015,L017,L020,L021,L022,L023,L024,L027,L029,L031,L032,L033,L034,L035,L037,L038,L041,L042,L043,L044,L045,L046,L047
sql_file_exts = .sql
dialect = tsql

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 80
indent_unit = tab
comma_style = leading
single_table_references = qualified
allow_scalar = False

[sqlfluff:rules:L010] # Keywords
capitalisation_policy = lower

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L030] # Function names
capitalisation_policy = lower
[sqlfluff:rules:L040] # Null & Boolean Literals
capitalisation_policy = lower

[sqlfluff:rules:L019]
comma_style = leading

[sqlfluff:rules:L028]
single_table_references = qualified




```
